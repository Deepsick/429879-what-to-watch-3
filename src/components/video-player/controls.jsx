import React, {Fragment, memo} from 'react';
import PropTypes from 'prop-types';
import {MAX_PERCENTAGE} from '../../const';
import {SECONDS_IN_MINUTE} from '../../const';

const getDragCoord = (duration, progress) => {
  return progress / duration * MAX_PERCENTAGE;
};

const getRunTime = (duration) => {
  const minutes = Math.floor(duration / SECONDS_IN_MINUTE);
  const seconds = Math.floor(duration - minutes * SECONDS_IN_MINUTE);

  return `${minutes}:${seconds}`;
};

const Controls = ({onExitButtonClick, onPlayButtonClick, isPlaying, onFullScreenButtonClick, duration, progress}) => (
  <Fragment>
    <button type="button" className="player__exit" onClick={onExitButtonClick}>Exit</button>

    <div className="player__controls">
      <div className="player__controls-row">
        <div className="player__time">
          <progress className="player__progress" value={getDragCoord(duration, progress)} max="100"></progress>
          <div className="player__toggler" style={{left: `${getDragCoord(duration, progress)}%`}}>Toggler</div>
        </div>
        <div className="player__time-value">{getRunTime(duration)}</div>
      </div>

      <div className="player__controls-row">
        <button type="button" className="player__play" onClick={onPlayButtonClick} >
          <svg viewBox="0 0 19 19" width="19" height="19">
            <use xlinkHref={isPlaying ? `#pause` : `#play-s`}></use>
          </svg>
          <span>{isPlaying ? `Pause` : `Play`}</span>
        </button>
        <div className="player__name">Transpotting</div>

        <button type="button" className="player__full-screen" onClick={onFullScreenButtonClick}>
          <svg viewBox="0 0 27 27" width="27" height="27">
            <use xlinkHref="#full-screen"></use>
          </svg>
          <span>Full screen</span>
        </button>
      </div>
    </div>
  </Fragment>
);

Controls.propTypes = {
  onExitButtonClick: PropTypes.func.isRequired,
  onPlayButtonClick: PropTypes.func.isRequired,
  onFullScreenButtonClick: PropTypes.func.isRequired,
  isPlaying: PropTypes.bool.isRequired,
  duration: PropTypes.number.isRequired,
  progress: PropTypes.number.isRequired,
};

export default memo(Controls);
