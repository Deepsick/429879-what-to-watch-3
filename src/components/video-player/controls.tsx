import * as React from 'react';

import history from '../../history';
import {MAX_PERCENTAGE, SECONDS_IN_MINUTE} from '../../const';

interface Props {
  isPlaying: boolean;
  duration: number;
  progress: number;
  onPlayButtonClick: () => void;
  onFullScreenButtonClick: () => void;
}

const getDragCoord = (duration, progress) => {
  return duration > 0 && progress / duration * MAX_PERCENTAGE;
};

const getRunTime = (duration, progress) => {
  const leftTime = duration - progress;
  const minutes = Math.floor(leftTime / SECONDS_IN_MINUTE);
  const seconds = Math.floor(leftTime - minutes * SECONDS_IN_MINUTE).toString().padStart(2, `0`);

  return `${minutes.toString().padStart(2, `0`)}:${seconds.toString().padStart(2, `0`)}`;
};

const handleExitButtonClick = (evt) => {
  evt.preventDefault();
  history.goBack();
};

const Controls: React.FunctionComponent<Props> = ({
  onPlayButtonClick,
  isPlaying,
  onFullScreenButtonClick,
  duration,
  progress
}: Props) => (
  <React.Fragment>
    <button type="button" className="player__exit" onClick={handleExitButtonClick}>Exit</button>

    <div className="player__controls">
      <div className="player__controls-row">
        <div className="player__time">
          <progress className="player__progress" value={getDragCoord(duration, progress)} max="100"></progress>
          <div className="player__toggler" style={{left: `${getDragCoord(duration, progress)}%`}}>Toggler</div>
        </div>
        <div className="player__time-value">{getRunTime(duration, progress)}</div>
      </div>

      <div className="player__controls-row">
        <button type="button" className="player__play" onClick={onPlayButtonClick} >
          <svg viewBox="0 0 19 19" width="19" height="19">
            <use xlinkHref={isPlaying ? `#pause` : `#play-s`}></use>
          </svg>
          <span>{isPlaying ? `Pause` : `Play`}</span>
        </button>
        <div className="player__name">Transpotting</div>

        <button type="button" className="player__full-screen" onClick={onFullScreenButtonClick}>
          <svg viewBox="0 0 27 27" width="27" height="27">
            <use xlinkHref="#full-screen"></use>
          </svg>
          <span>Full screen</span>
        </button>
      </div>
    </div>
  </React.Fragment>
);

export default React.memo(Controls);
