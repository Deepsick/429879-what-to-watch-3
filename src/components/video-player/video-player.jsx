import React, {PureComponent, createRef, Fragment} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {getMovies} from '../../reducer/data/selectors.js';
import Controls from './controls.jsx';

class VideoPlayer extends PureComponent {
  constructor(props) {
    super(props);
    this._videoRef = createRef();

    this.state = {
      isLoading: true,
      film: props.src || this._getFilmById(),
      isPlaying: props.isPlaying,
      progress: 0,
      duration: 0,
    };

    this._handlePlayButtonClick = this._handlePlayButtonClick.bind(this);
    this._handleFullScreenButtonClick = this._handleFullScreenButtonClick.bind(this);
  }

  _getFilmById() {
    const {films, id} = this.props;
    return films.find((film) => film.id === +id).video;
  }

  _handlePlayButtonClick() {
    const video = this._videoRef.current;
    const {isPlaying} = this.state;
    if (isPlaying) {
      video.pause();
    } else {
      video.play();
    }
  }

  _handleFullScreenButtonClick() {
    const video = this._videoRef.current;
    video.requestFullscreen();
  }

  componentDidMount() {
    const {film, muted, isPlaying} = this.state;

    const video = this._videoRef.current;

    video.src = film;
    video.muted = muted;

    video.oncanplaythrough = () => {
      this.setState({
        isLoading: false,
        duration: video.duration,
      });

      if (isPlaying) {
        video.play();
      } else {
        video.pause();
      }
    };

    video.onplay = () => {
      this.setState({
        isPlaying: true,
      });
    };

    video.onpause = () => {
      this.setState({
        isPlaying: false,
      });
    };

    video.ontimeupdate = () => this.setState({
      progress: Math.floor(video.currentTime),
    });
  }

  componentWillUnmount() {
    const video = this._videoRef.current;

    video.oncanplaythrough = null;
    video.onplay = null;
    video.onpause = null;
    video.ontimeupdate = null;
    video.src = ``;
  }

  render() {
    const {isControls, onExitButtonClick} = this.props;
    const {isPlaying, progress, duration, film} = this.state;
    const {poster} = film;

    return (
      <Fragment>
        <video
          ref={this._videoRef}
          width='100%'
          poster={poster || ``}
          className="player__video"
        ></video>
        {isControls &&
          <Controls
            onExitButtonClick={onExitButtonClick}
            onPlayButtonClick={this._handlePlayButtonClick}
            onFullScreenButtonClick={this._handleFullScreenButtonClick}
            isPlaying={isPlaying}
            duration={duration}
            progress={progress}
          />
        }

      </Fragment>
    );
  }
}

VideoPlayer.propTypes = {
  src: PropTypes.string,
  promo: PropTypes.shape({
    name: PropTypes.string.isRequired,
    poster: PropTypes.string.isRequired,
    preview: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    trailer: PropTypes.string.isRequired,
    video: PropTypes.string.isRequired,
    duration: PropTypes.number.isRequired,
    year: PropTypes.number.isRequired,
    cover: PropTypes.string.isRequired,
    bgColor: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    description: PropTypes.string.isRequired,
  }),
  muted: PropTypes.bool.isRequired,
  isPlaying: PropTypes.bool.isRequired,
  isControls: PropTypes.bool.isRequired,
  poster: PropTypes.string,
  onExitButtonClick: PropTypes.func,
};

const mapStateToProps = (state) => ({
  films: getMovies(state),
});

export default connect(mapStateToProps)(VideoPlayer);
