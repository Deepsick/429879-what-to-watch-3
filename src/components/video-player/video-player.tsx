import * as React from 'react';
import {connect} from 'react-redux';

import Controls from './controls';
import {getMovies} from '../../reducer/data/selectors';
import {Movie} from '../../types';

interface Props {
  films: Movie[];
  id: number;
  isControls: boolean;
}

interface State {
  isLoading: boolean;
  film: string;
  isPlaying: boolean;
  progress: number;
  duration: number;
}

class VideoPlayer extends React.PureComponent<Props, State> {
  private _videoRef: React.RefObject<HTMLVideoElement>;

  constructor(props) {
    super(props);
    this._videoRef = React.createRef();

    this.state = {
      isLoading: true,
      film: props.src || this._getFilmById(),
      isPlaying: true,
      progress: 0,
      duration: 0,
    };

    this._handlePlayButtonClick = this._handlePlayButtonClick.bind(this);
    this._handleFullScreenButtonClick = this._handleFullScreenButtonClick.bind(this);
  }

  componentDidMount() {
    const {film, isPlaying} = this.state;

    const video = this._videoRef.current;

    video.src = film;

    video.oncanplaythrough = () => {
      this.setState({
        isLoading: false,
        duration: video.duration,
      });

      if (isPlaying) {
        video.play();
      } else {
        video.pause();
      }
    };

    video.onplay = () => {
      this.setState({
        isPlaying: true,
      });
    };

    video.onpause = () => {
      this.setState({
        isPlaying: false,
      });
    };

    video.ontimeupdate = () => this.setState({
      progress: Math.floor(video.currentTime),
    });
  }

  componentWillUnmount() {
    const video = this._videoRef.current;

    video.oncanplaythrough = null;
    video.onplay = null;
    video.onpause = null;
    video.ontimeupdate = null;
    video.src = ``;
  }

  _getFilmById() {
    const {films, id} = this.props;
    return films.find((film) => film.id === +id).video;
  }

  _handlePlayButtonClick() {
    const video = this._videoRef.current;
    const {isPlaying} = this.state;
    if (isPlaying) {
      video.pause();
    } else {
      video.play();
    }
  }

  _handleFullScreenButtonClick() {
    const video = this._videoRef.current;
    video.requestFullscreen();
  }

  render() {
    const {isControls} = this.props;
    const {isPlaying, progress, duration} = this.state;

    return (
      <React.Fragment>
        <video
          ref={this._videoRef}
          width='100%'
          // poster={poster || ``}
          className="player__video"
        ></video>
        {isControls &&
          <Controls
            onPlayButtonClick={this._handlePlayButtonClick}
            onFullScreenButtonClick={this._handleFullScreenButtonClick}
            isPlaying={isPlaying}
            duration={duration}
            progress={progress}
          />
        }

      </React.Fragment>
    );
  }
}

const mapStateToProps = (state) => ({
  films: getMovies(state),
});

export {VideoPlayer};
export default connect(mapStateToProps)(VideoPlayer);
