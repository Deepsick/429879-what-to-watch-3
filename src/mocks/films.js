export const movies = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    picture: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    name: `Bohemian Rhapsody`,
    picture: `bohemian-rhapsody.jpg`,
  },
  {
    name: `Macbeth`,
    picture: `macbeth.jpg`,
  },
  {
    name: `Aviator`,
    picture: `aviator.jpg`,
  },
  {
    name: `We need to talk about Kevin`,
    picture: `we-need-to-talk-about-kevin.jpg`,
  },
  {
    name: `What We Do in the Shadows`,
    picture: `what-we-do-in-the-shadows.jpg`,
  },
  {
    name: `Revenant`,
    picture: `revenant.jpg`,
  },
  {
    name: `Johnny English`,
    picture: `johnny-english.jpg`,
  },
];

export const movie = {
  name: `The Grand Budapest Hotel`,
  genre: `drama`,
  year: 2014,
};
