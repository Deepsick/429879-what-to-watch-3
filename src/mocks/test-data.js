export const movie = {
  name: `The Grand Budapest Hotel`,
  genre: `drama`,
  year: 2014,
};

export const onMovieTitleClick = () => {};
export const onCardHover = () => {};
export const onMouseOut = () => {};
export const mockFunction = () => {};
export const hoverDelay = 1000;
export const isShown = true;
export const moviesCount = 8;
export const muted = true;
export const isPlaying = true;
export const genre = `drama`;
export const trailer = `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
export const isVideo = false;
export const mockBool = true;
export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`,
};

export const movies = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    picture: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    id: `1`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    name: `Bohemian Rhapsody`,
    picture: `bohemian-rhapsody.jpg`,
    id: `2`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    name: `Macbeth`,
    picture: `macbeth.jpg`,
    id: `3`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    name: `Aviator`,
    picture: `aviator.jpg`,
    id: `4`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
  {
    name: `We need to talk about Kevin`,
    picture: `we-need-to-talk-about-kevin.jpg`,
    id: `5`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    name: `What We Do in the Shadows`,
    picture: `what-we-do-in-the-shadows.jpg`,
    id: `6`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    name: `Revenant`,
    picture: `revenant.jpg`,
    id: `7`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
  {
    name: `Johnny English`,
    picture: `johnny-english.jpg`,
    id: `8`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  },
];

export const detailedMovies = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `drama`,
    id: `1`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody.jpg`,
    genre: `drama`,
    id: `2`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Macbeth`,
    poster: `macbeth.jpg`,
    genre: `drama`,
    id: `3`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Aviator`,
    poster: `aviator.jpg`,
    genre: `drama`,
    id: `4`,
    trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `We need to talk about Kevin`,
    poster: `we-need-to-talk-about-kevin.jpg`,
    genre: `drama`,
    id: `5`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `What We Do in the Shadows`,
    poster: `what-we-do-in-the-shadows.jpg`,
    genre: `drama`,
    id: `6`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Revenant`,
    poster: `revenant.jpg`,
    genre: `drama`,
    id: `7`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
  {
    name: `Johnny English`,
    poster: `johnny-english.jpg`,
    genre: `comedy`,
    id: `8`,
    trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    year: 2014,
    cover: `bg-the-grand-budapest-hotel.jpg`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    description: `
      In the 1930s, the Grand Budapest Hotel is a popular European ski resort,
      presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, 
      becomes Gustave's friend and protege.
  
      Gustave prides himself on providing first-class service to the hotel's guests,
      including satisfying the sexual needs of the many elderly women who stay there.
      When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a
      priceless painting and the chief suspect in her murder.
    `,
  },
];


