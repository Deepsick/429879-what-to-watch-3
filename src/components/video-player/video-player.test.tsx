import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {VideoPlayer} from './video-player';
import {movies, id, mockBool} from '../../mocks/test-data';

it(`Should render VideoPlayer component with controls correctly`, () => {
  const node = renderer.create(
      <VideoPlayer
        films={movies}
        isControls={mockBool}
        id={id}
      />, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

  expect(node).toMatchSnapshot();
});

it(`Should render VideoPlayer component without controls correctly`, () => {
  const node = renderer.create(
      <VideoPlayer
        films={movies}
        isControls={!mockBool}
        id={id}
      />, {
        createNodeMock: () => {
          return {};
        }
      }).toJSON();

  expect(node).toMatchSnapshot();
});
