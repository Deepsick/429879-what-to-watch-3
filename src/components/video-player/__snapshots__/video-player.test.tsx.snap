// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render VideoPlayer component with controls correctly 1`] = `
Array [
  <video
    className="player__video"
    width="100%"
  />,
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>,
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value={false}
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "false%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#pause"
          />
        </svg>
        <span>
          Pause
        </span>
      </button>
      <div
        className="player__name"
      >
        Transpotting
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`Should render VideoPlayer component without controls correctly 1`] = `
<video
  className="player__video"
  width="100%"
/>
`;
